core/src/main/scala/zio/logging/LogAnnotation.scala

```diff
@@ -0,0 +1,37 @@
+package zio.logging
+
+import scala.reflect.ClassTag
+
+/**
+ * A `LogAnnotation` describes a particular type of annotation applied to log
+ * lines.
+ */
+final case class LogAnnotation[A: ClassTag](name: String, initialValue: A, combine: (A, A) => A, render: A => String) {
+  self =>
+  def id: (String, ClassTag[A]) = (name, classTag)
+
+  /**
+   * The class tag of the annotation type, used for disambiguation purposes only.
+   */
+  def classTag: ClassTag[A] = implicitly[ClassTag[A]]
+
+  override def hashCode: Int = id.hashCode
+
+  override def equals(that: Any): Boolean = that match {
+    case that: LogAnnotation[_] => self.id == that.id
+    case _                      => false
+  }
+}
+
+object LogAnnotation {
+
+  /**
+   * The `Level` annotation keeps track of log levels.
+   */
+  val Level = LogAnnotation[LogLevel]("level", LogLevel.Info, (_, r) => r, _.render)
+
+  /**
+   * The `Name` annotation keeps track of logger names.
+   */
+  val Name = LogAnnotation[List[String]]("name", Nil, _ ++ _, _.mkString("."))
```

Comment:
Could be useful to add `CorrelationId` as `UUID`, with a combine that chooses the last, and a renderer that prints out the hex dash encoding (?).

Reply:
for correlationId it could be ```Option[UUID]``` because there are part of program that may run without correlation id e.g. startup logs, warmup, periodic log. Of course it depends how do we think about correlation id. what do you think?

Reply:
Yes, of course, `Option[UUID]`.